libcudacxx/include/cuda/std/__tuple_dir/ignore.h

```diff
@@ -0,0 +1,41 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX___TUPLE_IGNORE_H
+#define _LIBCUDACXX___TUPLE_IGNORE_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+struct __ignore_t
+{
```

Comment:
it may not be part of `std::ignore`, but I find these additions hugely useful in my own code:

```c++
  __ignore_t() = default;

  template <class... _Ts>
  _LIBCUDACXX_HIDE_FROM_ABI constexpr __ignore_t(_Ts&&...) noexcept
  {}

```

that way, i can define overloads like this:

```c++
template <class T>
  requires Fooable<T>
void do_foo(T& t);       // selected for Fooable types

void do_foo(__ignore_t); // fallback for all other types

Reply:
This looks interesting. However, wouldn't be a better idea to create a different utility for these cases? E. g. __sink_t or something similar?

Reply:
in my head, ignoring a function argument is the same kind of thing as ignoring a function's return value. i don't see a reason for a different utility. even in C++26, `std::ignore` will have an exposition-only type that users cannot utter in their code. i _think_ the extra constructor is a conforming extension.

Reply:
I would second @ericniebler that would be a fine addition and given that `__ignore_t` is a exposition only type we can conformingly add any constructor we want because the user has no official access to that type

Reply:
implemented